### PR TITLE
[Feature][Materialized-View] support materialized view on vectorized engine

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
# Proposed changes

Issue Number: close #xxx

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/functions/is_not_null.cpp [hunk 1]
```diff
@@ -56,7 +56,7 @@ class FunctionIsNotNull : public IFunction {
             auto* __restrict res_data = assert_cast<ColumnUInt8&>(*res_column).get_data().data();
 
             for (size_t i = 0; i < input_rows_count; ++i) {
-                res_data[i] = !src_data[i];
+                res_data[i] = src_data[i] == 0;
```
Comment (HappenLee) [inline be/src/vec/functions/is_not_null.cpp:59]: why is better ?

File: fe/fe-core/src/main/java/org/apache/doris/analysis/CreateMaterializedViewStmt.java [hunk 2]
```diff
@@ -399,6 +400,9 @@ private MVColumnItem buildMVColumnItem(FunctionCallExpr functionCallExpr) throws
                     mvColumnName = baseColumnName;
                 } else {
                     mvColumnName = mvColumnBuilder(functionName, baseColumnName);
+                    if (!functionChild0.getType().isStringType()) {
```
Comment (HappenLee) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/CreateMaterializedViewStmt.java:403]: why the origin do not have the logic

File: be/src/runtime/descriptors.cpp [hunk 3]
```diff
@@ -339,6 +336,10 @@ RowDescriptor::RowDescriptor(const DescriptorTbl& desc_tbl, const std::vector<TT
 
     init_tuple_idx_map();
     init_has_varlen_slots();
+
+    DCHECK(nullable_tuples.size() == _tuple_desc_map.size())
```
Comment (HappenLee) [inline be/src/runtime/descriptors.cpp:340]: seems no need move the DCHECK to here

File: fe/fe-core/src/main/java/org/apache/doris/alter/RollupJobV2.java [hunk 4]
```diff
@@ -150,6 +151,55 @@ public RollupJobV2(long jobId, long dbId, long tableId, String tableName, long t
...
+        }
+
+        IntSupplier colUniqueIdSupplier = new IntSupplier() {
+            public int pendingMaxColUniqueId = tbl.getMaxColUniqueId();
+
+            @Override
+            public int getAsInt() {
+                pendingMaxColUniqueId++;
+                return pendingMaxColUniqueId;
+            }
```
Comment (SWJTU-ZhangLei) [inline fe/fe-core/src/main/java/org/apache/doris/alter/RollupJobV2.java:172]: 1、now in rollup, we don't support light schema change, this is unnecessary. 
2、in the end, we also need update tbl max unique id . for example in  SchemaChangeJobV2.onFinished() , we do like this:
        //update max column unique id
        int maxColUniqueId = tbl.getMaxColUniqueId();
        for (Column column : tbl.getFullSchema()) {
            if (column.getUniqueId() > maxColUniqueId) {
                maxColUniqueId = column.getUniqueId();
            }
        }
        tbl.setMaxColUniqueId(maxColUniqueId);


3、i will refact the light schema change feature, here you can just delete this code about generating col_unique_id

**Comment by HappenLee** [comment]
LGTM

**Review by HappenLee — APPROVED** [review]

**Pull request merged**